cmake/modules/shared/CompilerSettings.cmake

```diff
@@ -16,7 +16,9 @@ add_compile_options(
   "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-Xfrontend -enable-upcoming-feature -Xfrontend ExistentialAny>"
   "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-Xfrontend -enable-upcoming-feature -Xfrontend InternalImportsByDefault>")
 
-# Platform specific defines.
+add_compile_definitions("SWT_BUILDING_WITH_CMAKE")
```

Comment:
I think that we should be proactive here and use the `$<$<COMPILE_LANGUAGE:Swift>:SWT_BUILDING_WITH_CMAKE>` spelling to limit it to Swift code rather than all languages.

Reply:
Is there a reason to limit it to Swift? I imagined it could be useful to our C++ target as well, although it isn't used there currently. But perhaps more importantly, I thought it could prevent unnecessary module rebuilds in case the `_Testing_Internals` module is built once _without_ this definition set and once _with_ it set.

Reply:
Ah, I see. I think that a comment to explain that would be nice then so that it doesn't seem like it was accidentally overapplied.